pootle/apps/pootle_translationproject/models.py

```diff
@@ -250,23 +250,33 @@ def init_store_from_template(self, template_store):
         pootle_path = ".".join(
             [posixpath.splitext(pootle_path)[0],
              template_store.filetype.extension.name])
-        store, __ = self.stores.get_or_create(
-            parent=self.create_parent_dirs(pootle_path),
-            pootle_path=pootle_path,
-            name=posixpath.basename(pootle_path))
-        return store
+        name = posixpath.basename(pootle_path)
+        if name in ["template.po", "templates.po"]:
```

Comment:
There should be a better way to check if this project uses a gnu-style layout. Note that you might have a non-gnu project which has files like `template.pot`.

Reply:
there is no longer gnu/nongnu etc - so this is the way to check that

re po/pot - the template(s) does have pot suffix - this is the name of the store that will be created

Reply:
Well, in the form we have a translation mapping preset for `gnu` style, can't we check if project's translation path mapping matches against it?

Reply:
no - not really the preset mapping could be (almost) anything as long as it has <language_code> somewhere and ends with `<ext>`

Reply:
sorry, i misunderstood the point about non-gnu templates.pot - in that case (which is a pretty corner-case) i think this is still the right way to deal with it

Reply:
The preset for `gnu` style is `/po/<language_code>.<ext>`. The preset is hardcoded, so I am pretty sure that whatever the translation path mapping the project uses, if it matches this preset we can tell the project layout is `gnu`. I think adding this check to the `if` condition will avoid that corner case I talked about.

Reply:
way too arbitrary - this is the correct solution - whenever it sees a template pot with name `template` (which is equiv of a lang code) it uses the lang code instead